### DISCUSSION TITLE
Insert keybindings not working as expected

### DISCUSSION THREAD

**Discussion by lyz-code** [LazyVim/LazyVim#2530] [Q&A] [answered]
I'm trying to create some bindings for the `<C-.*>` commands to be able to use them in insert mode such as:

```lua
keymap.set("i", "<C-h>", "<C-w>h", { desc = "Go to left window", remap = true })
keymap.set("i", "<C-j>", "<C-w>j", { desc = "Go to lower window", remap = true })
keymap.set("i", "<C-k>", "<C-w>k", { desc = "Go to upper window", remap = true })
keymap.set("i", "<C-l>", "<C-w>l", { desc = "Go to right window", remap = true })
```

I'm also trying to set an autocommand for `<c-s>` that works in normal mode but in insert it only leaves me in normal mode, so I have to type it twice

```lua
    keymap.set({ "i", "n" }, "<c-s>", function()
      -- Close all other windows
      vim.cmd("only")
      -- Save the buffer
      vim.cmd("w")
      -- Close the buffer
      vim.cmd("bd")
    end, { desc = "Save the current buffer and exit", silent = true })
```

Do you have any idea of what may be the reason?

Thanks!

**Comment by dpetka2001** [comment, marked as answer]
Try using `<Esc>` in the rhs like this `<Esc><C-w>h`. Your second keymap closes other windows, then saves the buffer and closes it for me even in `insert` mode. Don't have to press it twice.

**Reply by lyz-code** [reply]
Damn I should have catch that myself, sorry for wasting your time :(
